api/v1beta1/azuremachine_types.go

```diff
@@ -122,6 +122,10 @@ type AzureMachineSpec struct {
 	// DNSServers adds a list of DNS Server IP addresses to the VM NICs.
 	// +optional
 	DNSServers []string `json:"dnsServers,omitempty"`
+
+	// VMExtensions specifies a list of extensions to be added to the virtual machine.
+	// +optional
+	VMExtensions []VMExtension `json:"vmExtensions,omitempty"`
```

Comment:
I wonder if it would make sense to call this just "Extensions" since everything in here is pretty much in the context of a VM... but on the other hand I could see just "extensions" being too vague as it's used in other contexts 🤔 

wonder if others have thoughts?

Reply:
I saw that other pre-existing functions referring to VM extensions called it "VMExtensions" rather than "Extensions", such as the [VMExtensionSpecs() function](https://pkg.go.dev/sigs.k8s.io/cluster-api-provider-azure/azure/scope#MachineScope.VMExtensionSpecs) in `machine.go`. In order to keep things consistent, I called it `VMExtensions`. I'm not sure if other Azure resources have extensions, so maybe I'm being too specific here.

Reply:
I prefer having the "VM" prefix (disclaimer: I sort of really don't like the fact that we have to use such a generic word like "extension" to mean something very specific like "runs any arbitrary executable code on a VM at create time")

Reply:
> "runs any arbitrary executable code on a VM at create time"

that's only the case for the custom script extension (CSE), this would allow users to add any published Azure Marketplace extensions to the VMs, including custom script but also others that have very precise functionality. I also think the word "extension" is overloaded and vague but it's consistent with official Azure naming...

Reply:
How about `AzureExtension`?

Reply:
Is this doc only talking about CSE?

https://learn.microsoft.com/en-us/azure/virtual-machines/extensions/features-linux?tabs=azure-cli

(it regularly refers to "VM Extensions")

Reply:
that one is referring to all VM extensions, not just custom script
> Each Azure VM extension has a specific use case. Examples include:
> ...
> In addition to process-specific extensions, a Custom Script extension is available for both Windows and Linux virtual machines. The [Custom Script extension for Linux](https://learn.microsoft.com/en-us/azure/virtual-machines/extensions/custom-script-linux) allows any Bash script to be run on a VM

https://learn.microsoft.com/en-us/azure/virtual-machines/extensions/features-linux?tabs=azure-cli#use-cases-and-samples

Reply:
I think given above, `VMExtensions` makes the most sense